eda/src/autogluon/eda/auto/simple.py

```diff
@@ -1087,3 +1089,218 @@ def explain_rows(
+                "captures a non-linear relationship between the feature and the model's output.\n"
+                "* **Heterogeneity**: Check for instances where ICE lines have widely varying slopes, indicating different relationships "
+                "between the feature and the model's output for individual instances. This may suggest interactions between the feature "
+                "of interest and other features.\n"
+                "* **Outliers**: Look for any ICE lines that are very different from the majority of the lines. This may indicate potential "
+                "outliers or instances that have unique relationships with the feature of interest.\n"
+                "* **Confidence** intervals: If available, examine the confidence intervals around the PDP line. Wider intervals may indicate "
+                "a less certain relationship between the feature and the model's output, while narrower intervals suggest a more robust relationship.\n"
+                "* **Interactions**: By comparing PDPs and ICE plots for different features, you may detect potential interactions between features. "
+                "If the ICE lines change significantly when comparing two features, this might suggest an interaction effect.",
+                condition_fn=lambda _: show_help_text,
+            ),
+            PDPInteractions(features=features, fig_args=fig_args, sample=max_ice_lines, target=target, **chart_args),  # type: ignore
+            MarkdownSectionComponent(
+                f"The following variable(s) are categorical: {cats}. They are represented as the numbers in the figures above. "
+                f"Mappings are available in `state.pdp_id_to_category_mappings`. The`state` can be returned from this call via adding `return_state=True`.",
+                condition_fn=lambda _: len(id_to_category_mappings) > 0,
+            ),
+        ],
+    )
+
+    s = AnalysisState({"pdp_id_to_category_mappings": id_to_category_mappings})
+
+    return s if return_state else None
+
+
+def _validate_and_normalize_pdp_args(
+    train_data: pd.DataFrame,
+    features: Optional[Union[str, List[str]]] = None,
+    fig_args: Optional[Dict[str, Dict[str, Any]]] = None,
+    chart_args: Optional[Dict[str, Dict[str, Any]]] = None,
+    col_number_warning: int = 20,
+) -> Tuple[Dict[str, Any], Dict[str, Any], Optional[List[str]]]:
+    fig_args = get_empty_dict_if_none(fig_args)
+    chart_args = get_empty_dict_if_none(chart_args)
+
+    if features is not None:
+        if type(features) is not list:
+            features = [features]  # type: ignore
+        features_not_present = [f for f in features if f not in train_data.columns]
+        assert (
+            len(features_not_present) == 0
+        ), f"Features {', '.join(features_not_present)} are not present in train_data: {', '.join(train_data.columns)}"
+    if features is None and len(train_data.columns) > col_number_warning:
+        print("qqq")
```

Comment:
why do we need this print statement?

Reply:
Good catch - leftovers from debugging; removed.